sdk/core/core-xml/src/xml.browser.ts

```diff
@@ -10,6 +10,19 @@ if (!document || !DOMParser || !Node || !XMLSerializer) {
   );
 }
 
+// Policy to make our code Trusted Types compliant at running time.
+//   https://github.com/w3c/webappsec-trusted-types
+// We are calling DOMParser.parseFromString() to parse XML payload from Azure services.
+// The parsed DOM object is not exposed to outside. Scripts are disabled when parsing
+// according to the spec.  There are no HTML/XSS security concerns on the usage of
+// parseFromString() here.
+let ttPolicy: Pick<TrustedTypePolicy, "createHTML"> | undefined;
+if (typeof self.trustedTypes !== "undefined") {
+  ttPolicy = self.trustedTypes.createPolicy("@azure/core-xml#xml.browser", {
+    createHTML: (s) => s,
```

Comment:
What's the motivation for doing this, just to limit the types of things that can be created by the XML parser (e.g. disallow rogue scripts, that sort of thing in case someone manages to inject one into a service response)?

Reply:
If we don't do anything, Customers depending on our packages would get a runtime error from `parseFromString()` call when trust types is enabled.  Since we don't believe there's any HTML/XSS security concern in our code, a no-op policy to assert that we trust our code.  And customers can review this policy and add it to their allowed list.

Another option is to actually sanitize the input either using some NPM packages (e.g., dompurify), or a custom sanitizer.  But I don't think it bring any benefits while adds to bundle size/or adds complexity.

Reply:
That makes perfect sense, thanks!